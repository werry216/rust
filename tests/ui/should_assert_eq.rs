#![feature(plugin)]
#![plugin(clippy)]

#![deny(should_assert_eq)]

#[derive(PartialEq, Eq)]
struct NonDebug(i32);

#[derive(Debug, PartialEq, Eq)]
struct Debug(i32);

fn main() {
    assert!(1 == 2);
    assert!(Debug(1) == Debug(2));
    assert!(NonDebug(1) == NonDebug(1)); // ok

    test_generic(1, 2, 3, 4);
}

fn test_generic<T: std::fmt::Debug + Eq, U: Eq>(x: T, y: T, z: U, w: U) {
    assert!(x == y);
    assert!(z == w); // ok
}
