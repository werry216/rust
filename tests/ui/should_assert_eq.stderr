error: use `assert_eq` for better reporting
  --> $DIR/should_assert_eq.rs:13:5
   |
13 |     assert!(1 == 2);
   |     ^^^^^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/should_assert_eq.rs:4:9
   |
4  | #![deny(should_assert_eq)]
   |         ^^^^^^^^^^^^^^^^
   = note: this error originates in a macro outside of the current crate

error: use `assert_eq` for better reporting
  --> $DIR/should_assert_eq.rs:14:5
   |
14 |     assert!(Debug(1) == Debug(2));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro outside of the current crate

error: use `assert_eq` for better reporting
  --> $DIR/should_assert_eq.rs:21:5
   |
21 |     assert!(x == y);
   |     ^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro outside of the current crate

error: aborting due to 3 previous errors

